### ISSUE TITLE
Support GraphQL DataSource

### ISSUE THREAD

**Issue by tusharmath** [tailcallhq/tailcall#415]
Tailcall needs to support GraphQL data source. This could be supported in various ways:

- [Apollo Federation](https://www.apollographql.com/docs/federation/subgraph-spec/)
- [Open Federation](https://open-federation.org/)
- [GraphQL Fusion](https://open-federation.org/)
- Something else?

**Comment by sujeetsr** [comment]
Putting down some thoughts around how to do Apollo Federation...

## How Apollo Federation works:

- In a federated architecture, the individual GraphQL APIs are called **subgraphs**, and their individual schemas are composed into a **supergraph**.
- Each subgraph has to support the [subgraph spec](https://www.apollographql.com/docs/federation/subgraph-spec/).
- The individual subgraph schemas are composed together into a supergraph schema.
- Clients can query this schema by querying a **router**, which is the public access point for the supergraph.
- The router communicates with the individual subgraphs to resolve queries, and the individual subgraphs are not queryable by clients.

**Composition and Routing Options:**

- Apollo provides Apollo Studio for managed federation or the [Rover CLI](https://www.apollographql.com/docs/federation/quickstart/setup#install-the-rover-cli) for local composition. https://www.apollographql.com/docs/federation/quickstart/local-composition
- Apollo also provides the [Apollo Router](https://www.apollographql.com/docs/router/quickstart/) (a precompiled Rust binary) to server the composed schema.

## async-graphql Federation support:

[async-graphql](https://github.com/async-graphql/async-graphql) supports the subgraph spec, so an async-graphql server can act as a subgraph in a federated architecture. (dynamic schema is supported)
https://github.com/async-graphql/examples/tree/ed1e7f28e472b27d883092f660c7632942c4f314/federation

## Support Federation in Tailcall:

To enable Federation support in Tailcall, we can approach it in stages as follows:

**Milestone 1: Support Tailcall server acting as a subgraph in a federated architecture.**
- Do any additional steps (like entity resolver) that async-graphql does for federation
- Pass any federation related directives through to the final schema
- Compose multiple tailcall servers together with Apollo Rover and Router, and be able to serve a federated schema.

**Milestone 2: Develop our own composition library (and  possibly our own router).**
- Basically composition will involve creating a blueprint from multiple schemas, and this blueprint can output the supergraph schema. 
- We can develop our own router or continue to use Apollo router.
